extensions/resource-deployment/src/ui/deployClusterWizard/pages/deploymentProfilePage.ts

```diff
@@ -51,99 +62,83 @@ export class DeploymentProfilePage extends WizardPageBase<DeployClusterWizard> {
 				}
 			).withLayout({ width: '100%', height: '100%' });
 			const form = formBuilder.withLayout({ width: '100%' }).component();
-			this.loadCards().then(() => {
-				this._loadingComponent!.loading = false;
-			}, (error) => {
-				this.wizard.wizardObject.message = {
-					level: azdata.window.MessageLevel.Error,
-					text: localize('deployCluster.loadProfileFailed', "Failed to load the deployment profiles: {0}", error.message)
-				};
-				this._loadingComponent!.loading = false;
+			return view.initializeModel(form).then(() => {
+				return this.loadCards().then(() => {
```

Comment:
Dangling promise here. It'd probably just be easiest to move the error handler code into the out later (if you do make sure you make the inner call async so a thrown error is handled appropriately)